standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreUtils.java

```diff
@@ -359,4 +386,365 @@ public static String getPartitionValWithInvalidCharacter(List<String> partVals,
     }
     return colNames;
   }
+
+  /**
+   * validateName
+   *
+   * Checks the name conforms to our standars which are: "[a-zA-z_0-9]+". checks
+   * this is just characters and numbers and _
+   *
+   * @param name
+   *          the name to validate
+   * @param conf
+   *          hive configuration
+   * @return true or false depending on conformance
+   *              if it doesn't match the pattern.
+   */
+  public static boolean validateName(String name, Configuration conf) {
```

Comment:
Are these new methods moved from some other place? I can't see to find where these are coming from.

Reply:
Most of them come from metastore/MetaStoreUtils.